core/src/main/java/com/linecorp/armeria/server/CorsServerErrorHandler.java

```diff
@@ -50,49 +47,19 @@ public AggregatedHttpResponse renderStatus(@Nullable ServiceRequestContext ctx,
                                                @Nullable RequestHeaders headers,
                                                HttpStatus status, @Nullable String description,
                                                @Nullable Throwable cause) {
-
-        if (ctx == null) {
-            return serverErrorHandler.renderStatus(null, serviceConfig, headers, status, description, cause);
-        }
-
-        final CorsService corsService = ctx.findService(CorsService.class);
-        if (corsService == null) {
-            return serverErrorHandler.renderStatus(ctx, serviceConfig, headers, status, description, cause);
-        }
-
-        final AggregatedHttpResponse res = serverErrorHandler.renderStatus(ctx, serviceConfig, headers, status,
-                                                                           description, cause);
-
-        if (res == null) {
-            return serverErrorHandler.renderStatus(ctx, serviceConfig, headers, status, description, cause);
-        }
-
-        final CorsConfig corsConfig = corsService.config();
-        final ResponseHeaders updatedResponseHeaders = addCorsHeaders(ctx, corsConfig,
-                                                                      res.headers());
-
-        return AggregatedHttpResponse.of(updatedResponseHeaders, res.content());
+        return serverErrorHandler.renderStatus(ctx, serviceConfig, headers, status, description, cause);
     }
 
     @Nullable
     @Override
     public HttpResponse onServiceException(ServiceRequestContext ctx, Throwable cause) {
-        if (cause instanceof HttpResponseException) {
-            final HttpResponse oldRes = serverErrorHandler.onServiceException(ctx, cause);
-            if (oldRes == null) {
-                return null;
-            }
-            final CorsService corsService = ctx.findService(CorsService.class);
-            if (corsService == null) {
-                return oldRes;
-            }
-            return oldRes.recover(HttpResponseException.class, ex -> {
-                return ex.httpResponse()
-                         .mapHeaders(oldHeaders -> addCorsHeaders(ctx, corsService.config(), oldHeaders));
+        final CorsService corsService = ctx.findService(CorsService.class);
```

Comment:
Note: this was a problem before also, but preflight requests won't be handled correctly if an exception occurs before reaching `CorsService`.

Reply:
I've changed to apply CORS headers only for [simple requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#simple_requests
).

If there is an error in processing preflight requests, I don't think we need to set CORS headers.